ea-factories/src/factories/balance.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  ExecuteFactory,
+  Config,
+  DataResponse,
+  Account,
+  SequenceResponseData,
+} from '@chainlink/types'
+import objectPath from 'object-path'
+import { Validator, AdapterError } from '@chainlink/external-adapter'
+import { util } from '@chainlink/ea-bootstrap'
+
+const DEFAULT_DATA_PATH = 'addresses'
+const DEFAULT_CONFIRMATIONS = 6
+
+const WARNING_UNSUPPORTED_PARAMS = 'No Operation: this provider does not support'
+const ERROR_MISSING_ADDRESS = 'No Operation: address param is missing'
+
+export function addresses(accounts: Account[]) {
+  return accounts.reduce<string[]>((accumulator, current) => {
+    if (!current.warning) accumulator.push(current.address)
+    return accumulator
+  }, [])
+}
+
+export type IsSupported = (coin: string, chain: string) => boolean
+export type BalancesResponse = DataResponse<Account[], any>
+export type GetBalance = (account: Account, config: BalanceConfig) => Promise<BalancesResponse>
+export type GetBalances = (accounts: Account[], config: BalanceConfig) => Promise<BalancesResponse>
+
+export type BalanceConfig = Config & {
+  confirmations?: number
+  shouldOverwrite?: boolean
+  verbose?: boolean
```

Comment:
This functionality to trigger verbose responses on/off should be part of the framework and available to all adapters to use. Let's not expand the scope of this PR further, but please open a GitHub issue documenting this so we can improve the implementation in the future and bring it to all adapters.

Reply:
#211